wezterm-gui/src/tabbar.rs

```diff
@@ -179,6 +182,141 @@ impl TabBarState {
         &self.items
     }
 
+    // MacOS uses native buttons.
+    #[cfg(not(target_os = "macos"))]
+    fn fancy_window_decorations(
+        mouse_x: Option<usize>,
+        x: &mut usize,
+        config: &ConfigHandle,
+        items: &mut Vec<TabEntry>,
+        line: &mut Line,
+        colors: &TabBarColors,
+    ) {
+        let window_hide = parse_status_text(
+            &config.tab_bar_style.window_hide,
+            if config.use_fancy_tab_bar {
+                CellAttributes::default()
+            } else {
+                colors.window_hide().as_cell_attributes()
+            },
+        );
+        let window_hide_hover = parse_status_text(
+            &config.tab_bar_style.window_hide_hover,
+            if config.use_fancy_tab_bar {
+                CellAttributes::default()
+            } else {
+                colors.window_hide_hover().as_cell_attributes()
+            },
+        );
+
+        let window_maximize = parse_status_text(
+            &config.tab_bar_style.window_maximize,
+            if config.use_fancy_tab_bar {
+                CellAttributes::default()
+            } else {
+                colors.window_maximize().as_cell_attributes()
+            },
+        );
+        let window_maximize_hover = parse_status_text(
+            &config.tab_bar_style.window_maximize_hover,
+            if config.use_fancy_tab_bar {
+                CellAttributes::default()
+            } else {
+                colors.window_maximize_hover().as_cell_attributes()
+            },
+        );
+
+        let window_close = parse_status_text(
+            &config.tab_bar_style.window_close,
+            if config.use_fancy_tab_bar {
+                CellAttributes::default()
+            } else {
+                colors.window_close().as_cell_attributes()
+            },
+        );
+        let window_close_hover = parse_status_text(
+            &config.tab_bar_style.window_close_hover,
+            if config.use_fancy_tab_bar {
+                CellAttributes::default()
+            } else {
+                colors.window_close_hover().as_cell_attributes()
+            },
+        );
+
+        let window::FancyWindowDecorations {
+            is_left,
+            remove_hide_button,
+            remove_maximize_button,
+            ..
+        } = if cfg!(any(windows, target_os = "macos")) {
+            // Ignore changes to config on windows and macos
+            window::FancyWindowDecorations::default()
+        } else {
+            config.fancy_window_decorations.clone()
+        };
+
+        let button_order = if is_left {
+            ['X', '.', '-']
+        } else {
+            ['.', '-', 'X']
+        };
```

Comment:
As a stylistic nit, using characters for these is prone to typos and the need for a catch-all in the match statement below.

I'd suggest using a little enum for this:

```rust
enum Button {
   Hide,
   Maximize,
   Close,
}

let button_order = if is_left {
   [Button::Close, Button::Hide, Button::Maximize]
} else {
    [Button::Hide, Button::Maximize, Button::Close]
};
```

and then using that below; it would also mean that you wouldn't need the comments alongside those cases to explain that `'.'` means hide and so on.